Bottom: 813e1c0119d42fc4ad6948c4b7e8c12ae999b10a
Top:    c0f45c3c6915bba9168829f2458804363b5d338f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2015-01-11 23:39:32 +0000

PCB+GL: Fix GL headers for Win32


---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index 732cef5..01a4479 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -32,12 +32,18 @@
 #include <math.h>
 #include <assert.h>
 
-/* The Linux OpenGL ABI 1.0 spec requires that we define
- * GL_GLEXT_PROTOTYPES before including gl.h or glx.h for extensions
- * in order to get prototypes:
- *   http://www.opengl.org/registry/ABI/
- */
-#define GL_GLEXT_PROTOTYPES 1
+#ifdef WIN32
+#   define WIN32_LEAN_AND_MEAN 1
+#   include <windows.h>
+#else
+    /* The Linux OpenGL ABI 1.0 spec requires that we define
+     * GL_GLEXT_PROTOTYPES before including gl.h or glx.h for extensions
+     * in order to get prototypes:
+     *   http://www.opengl.org/registry/ABI/
+     */
+#   define GL_GLEXT_PROTOTYPES 1
+#endif
+
 #ifdef HAVE_OPENGL_GL_H
 #   include <OpenGL/gl.h>
 #else
